c/tests/testlib.h

```diff
@@ -55,6 +55,22 @@ void parse_individuals(const char *text, tsk_individual_table_t *individual_tabl
 void unsort_edges(tsk_edge_table_t *edges, size_t start);
 
 void assert_arrays_almost_equal(tsk_size_t len, double *a, double *b);
+#define ASSERT_ARRAYS_ALMOST_EQUAL(len, a, b)                                           \
+    do {                                                                                \
+        tsk_size_t j;                                                                   \
+                                                                                        \
+        for (j = 0; j < len; j++) {                                                     \
+            CU_ASSERT_DOUBLE_EQUAL(a[j], b[j], 1e-9);                                   \
+        }                                                                               \
+    } while (0)
+#define ASSERT_ARRAYS_EQUAL(len, a, b)                                                  \
+    do {                                                                                \
+        tsk_size_t j;                                                                   \
+                                                                                        \
+        for (j = 0; j < len; j++) {                                                     \
+            CU_ASSERT_EQUAL(a[j], b[j]);                                                \
+        }                                                                               \
+    } while (0)
```

Comment:
I've added these because it makes failures much easier to debug. For example, with the original `assert_arrays_almost_equal` function, we see this type of thing when the assertion fails:

```
  Test: test_paper_ex_two_site ...FAILED
    1. ../tests/testlib.c:975  - CU_ASSERT_DOUBLE_EQUAL(a[j],b[j],1e-9)
```

Note the source file/line number reference. Now, with this new macro `ASSERT_ARRAYS_ALMOST_EQUAL`, we see this:

```
  Test: test_paper_ex_two_site ...FAILED                                                                                                                       
    1. ../tests/test_stats.c:2032  - CU_ASSERT_DOUBLE_EQUAL(result[j],truth_one_set[j],1e-9)
```

I now see the exact line where my assertion fails.

I see this as a quality of life improvement, but I'm happy to remove it if you think it doesn't fit here.

Reply:
Yep, very nice. I'm sure I already did something equivalent, but it must not have been merged yet.